File: python/problems/medium/implement_trie_prefix_tree.py
from __future__ import annotations


class Trie:
    def __init__(self):
        pass

    def insert(self, word: str) -> None:
        raise NotImplementedError

    def search(self, word: str) -> bool:
        raise NotImplementedError

    def startsWith(self, prefix: str) -> bool:
        raise NotImplementedError

File: rust/src/problems/implement_trie_prefix_tree.rs
//! Implement Trie (Prefix Tree) (Medium).

pub struct Trie;

impl Trie {
    pub fn new() -> Self {
        unimplemented!("implement-trie-prefix-tree")
    }

    pub fn insert(&self, word: String) {
        unimplemented!("implement-trie-prefix-tree")
    }

    pub fn search(&self, word: String) -> bool {
        unimplemented!("implement-trie-prefix-tree")
    }

    pub fn starts_with(&self, prefix: String) -> bool {
        unimplemented!("implement-trie-prefix-tree")
    }
}

pub(crate) fn run_case() {
    let trie = Trie::new();
    trie.insert("apple".into());
    assert!(trie.search("apple".into()));
    assert!(trie.starts_with("app".into()));
}

#[cfg(test)]
mod tests {
    #[test]
    fn representative() {
        super::run_case();
    }
}
